packages/jellyfish-transaction/__tests__/script/dftx/dftx_pool/PoolSwap.test.ts

```diff
@@ -79,4 +77,93 @@ describe('Composable', () => {
 
     expect(buffer.toBuffer().toString('hex')).toStrictEqual(data)
   })
+
+  it('should bi-directional with two digits precision for 1.92', () => {
+    const poolSwapTwoDecimalPrecision = {
+      ...poolSwap,
+      maxPrice: new BigNumber('1.92')
+    }
+
+    const composable = new CPoolSwap(poolSwapTwoDecimalPrecision)
+    const buffer = new SmartBuffer()
+    composable.toBuffer(buffer)
+
+    const hex = buffer.toBuffer().toString('hex')
+
+    const buffer2 = SmartBuffer.fromBuffer(Buffer.from(hex, 'hex'))
+    const composable2 = new CPoolSwap(buffer2)
+
+    expect(composable2.toObject()).toStrictEqual(poolSwapTwoDecimalPrecision)
+  })
+
+  it('should bi-directional with seven digits precision for 0.00000003', () => {
+    const poolSwapEightnDecimalPrecision = {
+      ...poolSwap,
+      maxPrice: new BigNumber('0.00000003')
+    }
+
+    const composable = new CPoolSwap(poolSwapEightnDecimalPrecision)
+    const buffer = new SmartBuffer()
+    composable.toBuffer(buffer)
+
+    const hex = buffer.toBuffer().toString('hex') // "17a914c34ca9c54dc87e7e875b212ec6ba0704be3de587870000d6117e0300000017a914c34ca9c54dc87e7e875b212ec6ba0704be3de587870200000000000000000300000000000000"
+
+    const buffer2 = SmartBuffer.fromBuffer(Buffer.from(hex, 'hex'))
+    const composable2 = new CPoolSwap(buffer2)
+
+    expect(composable2.toObject()).toStrictEqual(poolSwapEightnDecimalPrecision)
+  })
+
+  it('should bi-directional with two digits precision for 2.0', () => {
+    const poolSwapFromDivision = {
+      ...poolSwap,
+      maxPrice: new BigNumber(100 / 50)
+    }
+
+    const composable = new CPoolSwap(poolSwapFromDivision)
+    const buffer = new SmartBuffer()
+    composable.toBuffer(buffer)
+
+    const hex = buffer.toBuffer().toString('hex')
+    const buffer2 = SmartBuffer.fromBuffer(Buffer.from(hex, 'hex'))
+    const composable2 = new CPoolSwap(buffer2)
+
+    expect(composable2.toObject()).toStrictEqual(poolSwapFromDivision)
+  })
+
+  it('should bi-directional with UInt64MAX', () => {
+    const poolSwapUint64MAX = {
+      ...poolSwap,
+      maxPrice: new BigNumber('18446744073709551615')
```

Comment:
I haven't read through thoroughly, this suppose to have decimal point in between? @Jouzo